backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CloneVmCommand.java

```diff
@@ -134,21 +132,24 @@ protected void init() {
 
     @Override
     protected void executeVmCommand() {
-        getParameters().setStage(CloneVmParameters.CloneVmStage.CREATE_VM_SNAPSHOT);
-        Guid snapshotId = createVmSnapshot();
-        setSnapshotId(snapshotId);
-        setSucceeded(snapshotId != null);
+        if (isAllVmDisksSupportSnapshots()) {
+            getParameters().setStage(CloneVmParameters.CloneVmStage.CREATE_VM_SNAPSHOT);
+            Guid snapshotId = getVmSnapshotToClone();
+            initDisks(snapshotId);
+            setSucceeded(snapshotId != null);
+        } else {
+            getParameters().setStage(CloneVmParameters.CloneVmStage.COPY_DISKS);
+            setSucceeded(true);
+        }
+
     }
 
-    private void setSnapshotId(Guid snapshotId) {
+    private void initDisks(Guid snapshotId) {
         getParameters().setSourceSnapshotId(snapshotId);
         diskImagesFromConfiguration = null;
         diskInfoDestinationMap = new HashMap<>();
         fillDisksToParameters();
-        storageToDisksMap =
-                ImagesHandler.buildStorageToDiskMap(getImagesToCheckDestinationStorageDomains(),
-                        diskInfoDestinationMap);
-        getParameters().setStorageToDisksMap(storageToDisksMap);
+        getParameters().setStorageToDisksMap(getStorageToDisksMap());
```

Comment:
do we still use the map from the parameters?

Reply:
No with the new attitude of setting the map in getStorageToDisksMap()